modules/programs/lsd.nix

```diff
@@ -45,6 +45,25 @@ in {
         for supported values.
       '';
     };
+
+    colors = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = {
+        size = {
+          none = "grey";
+          small = "yellow";
+          large = "dark_yellow";
+        };
+      };
+      description = ''
+        Configuration written to
+        {file}`$XDG_CONFIG_HOME/lsd/colors.yaml`. See
+        <https://github.com/lsd-rs/lsd/tree/v1.0.0#color-theme-file-content>
+        for supported colors.
+        Make sure you have set the `settings.color.theme = "custom";`.
```

Comment:
Perhaps simply add

```
programs.lsd = mkIf (cfg.colors != { }) { settings.color.theme = "custom"; };
```

to the config section? Then the user shouldn't be forced to explicitly set the option.